_posts/en/newsletters/2019-01-08-newsletter.md

```diff
@@ -0,0 +1,168 @@
+    [discussed][rm codesep] is whether the `OP_CODESEPARATOR` opcode
+    should be disabled in a script update that supports MAST (e.g. via
+    Taproot).  That opcode is not in common use, but if you plan to use
+    it in future Script versions, you should comment on the thread.
+
+- **Cross-chain LN as an options contract:** pseudonymous LN
+  contributor ZmnSCPxj started a thread on the Lightning-Dev mailing
+  list pointing out that users could abuse payments that cross
+  currencies to create almost free [short-term options contracts][] by
+  delaying payment settlement.  A [previous thread][cjp risk] by Corné
+  Plooy in May 2018 described the same thing.
+  
+    For example, Mallory learns that Bob is willing to route payments
+    from Bitcoin to Litecoin, so she sends a payment from one of her
+    Bitcoin nodes through Bob to one of her Litecoin nodes.  If this
+    were a normal payment, she'd settle it immediately by releasing the
+    preimage for the payment's hashlock---but instead her node delays
+    for 24 hours waiting for the exchange rate to change.  If the
+    exchange rate increases in Litecoin's favor, Mallory settles the
+    payment and receives litecoin today at yesterday's exchange rate.
+    If the exchange rate stays the same or increases in Bitcoin's favor,
+    Mallory causes the payment to fail and gets her bitcoin back.  Since
+    no fees are charged for failed payments, Mallory received an
+    opportunity to temporarily lock-in the price of Litecoin for nothing
+    but the cost of owning the bitcoins Mallory would've traded.
+
+    There currently aren't any known cross-currency LN nodes, but the
+    availability of this trick means that future such nodes could be
+    abused for speculation rather than payment routing.  If this turns
+    out to be a real problem and if an acceptable solution isn't found,
+    it may be the case that payment channel networks for different
+    currencies will be isolated from each other.
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][core commits],
+[LND][lnd commits], [C-lightning][cl commits], and [libsecp256k1][secp
+commits].*
+
+- [Bitcoin Core #14565][] significantly improves the error handling for
+  the `importmulti` RPC and will return a `warnings` field for each
+  attempted import with an array of strings describing any problems with
+  the that import (but only if there were any problems).
+
+- [Bitcoin Core #14811][] updates the [getblocktemplate][rpc
+  getblocktemplate] RPC to require that the segwit flag be passed.  This
+  helps prevent miners from accidentally not using segwit, which reduces
+  their fee income.  See [Newsletter #20][] for a recent instance where
+  this may have happened to a large mining pool.
+
+- [Bitcoin Core #14336][] changes the system call (syscall) Bitcoin Core
+  uses to handle network connections and other resources on platforms
```

Comment:
That user only encountered those problems because they were running with more than the default maximum number of connections:

```
229 2018-07-31T19:58:47  <sipa> how many connwctions do you have?
230 2018-07-31T20:01:26  <ossifrage> sipa, 160
```

IIUC, the lack of file descriptors never caused problems with the database---the user-visible problem was that sometimes the user couldn't previously get to their configured maximum number of connections, as we described in Newsletter 8:

> Bitcoin Core #13925: increases the maximum number of file descriptors Bitcoin Core’s internal database can use, which can allow more file descriptors to be used for network connections. If you’ve modified Bitcoin Core to accept more than 117 incoming connections, you may see an additional increase in the number of connections after upgrading past this merge. (Note: we don’t recommend increasing the default unless you have a special need.)

We also previously discussed some technical aspects of this issue: https://github.com/bitcoinops/bitcoinops.github.io/pull/58#issuecomment-414384299

In short, I think the description in the newsletter now describes the user-visible changes, which are what I think is most important to our readers, rather than going into the technical details (which are still available to anyone who clicks the link to the PR).

I don't think I'll be around to see your reply to this issue before publication time, so feel free to (1) modify the text as you think is appropriate or (2) drop the bullet point from this newsletter and we can discuss it and optionally put it in next week's newsletter.